lektor/reporter.py

```diff
@@ -10,6 +12,14 @@
 
 _reporter_stack = LocalStack()
 _build_buffer_stack = LocalStack()
+_change_stream_reporters = WeakSet()
+
+
+def report_artifact_built(artifact, is_current):
+    if is_current:
+        return
+    for reporter in _change_stream_reporters:
+        reporter.change_stream.put(artifact)
```

Comment:
Can `_change_stream_reporters` and `report_artifact_built` be properties/methods of `Reporter` rather than globals?

Reply:
`report_artifact_built` and `_change_stream_reporters` *broadcasts* to all threaded reporters, while `Reporter` is stored on a `werkzeug.local.LocalStack`, which is specific for each thread.

Reply:
Sure, I changed it to a class attribute